### DISCUSSION TITLE
Plans for the future

### DISCUSSION THREAD

**Discussion by csirikk** [the-marenga/sf-scrapbook-helper#53] [General]
I'm rather interested in this project and i'm wondering whether you have plans to expand this tool from purely scrapbook based to other parts of sfgame being automated?

For example fortress attacks, various dungeons, pets, arena manger, maybe even quests (which i'm not sure how would work with the new 'expeditions').

**Comment by the-marenga** [comment]
I have thought about that a lot. My main reason, why I am hesitant to expand this tool currently is the idea of the MFBot to become open source: ["we’re currently looking into how we can pull off getting the code into Open Source." ](https://www.mfbot.de/en/status-update-2024-net-6-nightly/)

If they actually do that, then it makes a lot more sense to just expand that tool, instead of writing all the "which item is better" "which quest is better" logic again from scratch. 

Most of the people, that are interested in the sf-scrapbook-helper are already using the MFBot and it does not really make sense to have two tools running, that both work on the same aspects of the game. The only real chance my tool would have long term in these aspects is, if I at least match the feature set of the MFBot, which would be quite a lot of work.

Another thing is, that IIRC github does not like cheats/bots on their platform, so if I expand the automated functionality, there is a chance this gets removed. I would have to look back into their guidelines though to make sure I am not mixing something up.

I have however already implemented all the server communication logic to send and parse the expedition, pet, fortresss etc. stuff in the underlying API [here](https://github.com/the-marenga/sf-api).

I am on holliday for the ~next two weaks, so for the short term, there is just a small release planned. After that I am planning to fix all the bigger bugs and improve the UI and once that is done, I will check back on the MFBot and see, if the open sourcing initiative of the MFBot went anywhere. If that went nowhere, I will decide how I want to continue this project. 

If anyone else knows how to write Rust though, I would of course happily accept PRs, that expand the functionality of this tool at any time
